v2/lints/cabf_br/lint_ocsp_id_pkix_ocsp_nocheck_ext_not_included_test.go

```diff
@@ -0,0 +1,68 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/test"
+)
+
+func Test_SMIME_CAWrong(t *testing.T) {
+	inputPath := "ocspidpkixocspnocheckextnotincluded_SMIME_CA_wrong.pem"
+	expected := lint.Warn
+	out := test.TestLint("e_ocsp_id_pkix_ocsp_nocheck_ext_not_included", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
+
+func Test_SMIME_CACorrect(t *testing.T) {
+	inputPath := "ocspidpkixocspnocheckextnotincluded_SMIME_CA_correct.pem"
+	expected := lint.NA
+	out := test.TestLint("e_ocsp_id_pkix_ocsp_nocheck_ext_not_included", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
+
+
+func Test_TLS_CAWrong(t *testing.T) {
+	inputPath := "ocspidpkixocspnocheckextnotincluded_TLS_CA_wrong.pem"
+	expected := lint.Error
+	out := test.TestLint("e_ocsp_id_pkix_ocsp_nocheck_ext_not_included", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
+
+func Test_TLS_CACorrect(t *testing.T) {
+	inputPath := "ocspidpkixocspnocheckextnotincluded_TLS_CA_correct.pem"
+	expected := lint.NA
+	out := test.TestLint("e_ocsp_id_pkix_ocsp_nocheck_ext_not_included", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
+
+func TestDelegatedResponder(t *testing.T) {
+	inputPath := "ocspidpkixocspnocheckextnotincluded_delegated_responder.pem"
+	expected := lint.Pass
+	out := test.TestLint("e_ocsp_id_pkix_ocsp_nocheck_ext_not_included", inputPath)
+	if out.Status != expected {
+		t.Errorf("%s: expected %s, got %s", inputPath, expected, out.Status)
+	}
+}
```

Comment:
This isn't a hard block, but a number of our newer tests uses a loop, which makes it a lot easier to distinguish the test case / expectations from the boiler plate.

https://github.com/zmap/zlint/blob/aa4e2619db00dbdea768e9afa70262f0d6af3417/v2/lints/apple/lint_ct_sct_policy_count_unsatisfied_test.go#L31 is an example, if you'd like to model, which makes it slight easier to maintain and refactor.

Reply:
@RufusJWB Any thoughts on this?

Reply:
@sleevi I followed your proposal and refactored test test into a loop. To make this whole thing more re-useable and minimize the amount of duplicated code, I moved the definition of the `TestCase` struct and of the `RunTest` func into https://github.com/RufusJWB/zlint/blob/master/v2/test/helpers.go . Now they can be re-used wherever needed in the code.

Reply:
I rolled back the refactoring and puit it into a different branch https://github.com/RufusJWB/zlint/blob/474-refactor-tests/v2/test/helpers.go Later we can merge it in a separate PR